### PR TITLE
Improve sounds

### LINKED ISSUE #8: Improve sounds
At the moment, the game has no sounds. Add sounds for:

* Player shooting
* Invaders shooting
* Invaders moving
* Background tune (should not play until the game has started by the user)

### PR DESCRIPTION
**Pull request by Matt-Warnock**
I already apoligise for how nuts some of this code is. I've needed to concider that the tick is still runing as animation after game win or lose is desirble. My solution was to stop playerStatus method after the game finishes but to keep the gameover/win message on screen.

### CODE REVIEW

File: game.js [hunk 1]
```diff
@@ -23,7 +24,28 @@ class Game {
...
   }
 
+  run() {
+    let tick = () => {
+      this.update();
+      this.draw();
+      if (this.gameInProgress) {
+        this.playerStatus();
+        this.sound.backgroundMusic();
+        this.sound.invaderStepMute(false);
```
Comment (octopusinvitro) [inline game.js]: If you are not using this `invaderStepMute(false)` anywhere else, I recommend you to absorbe its logic inside of `backgroundMusic()` and maybe rename it to `startSound()` `stopSound()`. It's a bad practice to pass a boolean as an argument of a method because it violates the Single Responsibility Principle, as in, the method is doing two things.

Also when you stop a sound you also have to set `currentTime` to zero, so that it starts from the beginning the next time you play it.

```js
  startSound() {
    this.sounds.backgroundSong.play();
    this.sounds.invaderMove.muted = false;
  }

  stopSound() {
    this.sounds.backgroundSong.pause();
    background.backgroundSong.currenTime = 0;
    this.sounds.invaderMove.muted = true;
  }
```

Why are you using mute instead of pause like in the others?

File: game.js [hunk 2]
```diff
@@ -23,7 +24,28 @@ class Game {
     });
   }
 
+  run() {
+    let tick = () => {
+      this.update();
+      this.draw();
+      if (this.gameInProgress) {
+        this.playerStatus();
```
Comment (octopusinvitro) [inline game.js]: Why did you change this name? :thinking:  `checkStatus()` sounded like a command method. With `playerStatus()` I was expecting it to return something.

File: sound.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  backgroundMusic() {
+    this.sounds.backgroundSong.play();
+  }
+
+  stopBackgroundMusic() {
+    this.sounds.backgroundSong.pause();
+  }
+
+  lazerBlast() {
+    this.sounds.playerLaser.cloneNode(true).play();
```
Comment (octopusinvitro) [inline sound.js]: This is very confusing and a mental overload. All these methods are called X, yet they play sound Y. Why not unify method name and sound name so that it is more readable:

```js
  lazerBlast() {
    this.sounds.lazerBlast.cloneNode(true).play();
  }

  shipExplosion() {
    this.sounds.shipExplosion.cloneNode(true).play();
  }

  ... etc.
```

**Review by octopusinvitro — COMMENTED** [review]
Make sure that all the files you added to the PR are the files you need. For example you are not using `fastinvader4.mp3` or `player_laser5` but they are in the PR.

Also if you used music from someone, remember to mention it in the footer :+1:

File: game.js (hunk 1 above)
Comment (Matt-Warnock) [reply game.js]: because I belive pause will only stop the currant one from playing

Comment (Matt-Warnock) [reply game.js]: also because Invaders update() will continue to run after gameover for animation, that will continuly command .play(). I feel muting is a less intrusive way to prevent the user from hearing it.

File: game.js (hunk 2 above)
Comment (Matt-Warnock) [reply game.js]: thought it read better, I'll change it back
